pkg/telemetry/telemetry.go

```diff
@@ -0,0 +1,281 @@
+// This file and its contents are licensed under the Apache License 2.0.
+// Please see the included NOTICE for copyright information and
+// LICENSE for a copy of the license.
+
+package telemetry
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jackc/pgtype"
+
+	"github.com/prometheus/client_golang/prometheus"
+	io_prometheus_client "github.com/prometheus/client_model/go"
+
+	"github.com/timescale/promscale/pkg/log"
+	"github.com/timescale/promscale/pkg/pgmodel/common/schema"
+	"github.com/timescale/promscale/pkg/pgmodel/model/pgutf8str"
+	"github.com/timescale/promscale/pkg/pgxconn"
+)
+
+type engine struct {
+	uuid [16]byte
+	conn pgxconn.PgxConn
+
+	stop chan struct{}
+
+	metrics sync.Map
+}
+
+type Telemetry interface {
```

Comment:
IMO more fitting name here is `Engine`. 

Would reduce stutter: `telemetry.Engine` vs `telemetry.Telemetry` 
and it seems you call it engine in var names everywhere.